suite-native/biometrics/src/useBiometrics.tsx

```diff
@@ -30,45 +29,43 @@ export const useBiometrics = () => {
     const { isBiometricsOptionEnabled } = useIsBiometricsEnabled();
     const { isUserAuthenticated, setIsUserAuthenticated } = useIsUserAuthenticated();
     const { setIsBiometricsOverlayVisible } = useIsBiometricsOverlayVisible();
-    const { isBiometricsAuthenticationCanceled, setIsBiometricsAuthenticationCanceled } =
-        useIsBiometricsAuthenticationCanceled();
     const appState = useRef(AppState.currentState);
     const [appStateVisible, setAppStateVisible] = useState(AppState.currentState);
+    const [isBiometricsAuthenticationAllowed, setIsBiometricsAuthenticationAllowed] =
+        useState(true);
     const goneToBackgroundAtTimestamp = useRef<null | number>(null);
 
-    const handleAuthentication = useCallback(async () => {
-        // Stop the authentication flow if the user leaves the app.
-        if (appState.current !== 'active' && Platform.OS === 'android') {
-            LocalAuthentication.cancelAuthenticate();
+    const checkAuthentication = useCallback(async () => {
+        // Stop the authentication flow if the user is not active
+        if (appState.current !== 'active') {
+            // on android app is either active or in background, canncel auth if not active
+            if (Platform.OS === 'android') {
+                LocalAuthentication.cancelAuthenticate();
+            }
 
             return;
         }
 
         if (
```

Comment:
What about moving this condition to useEffect where this method checkAuthentication is called? It would be easier to read the code (no extra dependencies in the useEffect) and prevent calling `LocalAuthentication.cancelAuthenticate();` if this condition is not met. Now it can call LocalAuthentication.cancelAuthenticate(); on background even if `isBiometricsOptionEnabled === false`.

Reply:
Updated in b0d80b2974. Please test on device 🙏